src/runner/runESLint.js

```diff
@@ -1,32 +1,6 @@
-const { ESLint } = require('eslint');
+const { ESLint, loadESLint } = require('eslint');
 const getESLintOptions = require('../utils/getESLintOptions');
 
-let FlatESLint;
```

Comment:
why is all this important compat code being removed?

Reply:
As I'd mentioned in the PR description, this code is all superseded by `loadESLint` which was added in eslint 8.57.0 (and is exactly the API I'd originally requested to implement in https://github.com/eslint/eslint/issues/17129).  This code was only ever here because we weren't able to get the API added in eslint at the time.

The effect of removing the code is that the flat config won't work with versions of jest-runner-eslint after this change if eslint is pinned to a version between 8.41.0 and 8.56.0.

Even if we kept the compat code, we'd want to use the new `loadESLint` since it's the officially supported mechanism for doing this.  What we could do us use `loadESLint` if it exists, and fall back to the compat code if it doesn't.  However, the use case for moving eslint to 8.57.0, but locking jest-runner-eslint to a lower version, seems to me to make very little sense -- enough that I doubt it's worth maintaining the added complexity.  Given the flat config has been experimental anyways and was not guaranteed to be stable, I don't even think this should actually be considered a breaking change, although it _technically_ is.

We can add back the compat code if you felt strongly, but I seriously doubt there's any practical benefit of having it.

Reply:
That makes it a breaking change, so we'd want to keep the compat code regardless, and that's fine to use loadESLint and fall back.

Avoiding breaking changes has the largest possible practical benefit, higher than any features or bug fixes.

Reply:
Fair enough -- although it's worth noting that internals in ESLint [have been renamed](https://github.com/eslint/eslint/pull/17914) over those versions, were never guaranteed to be stable, and likely the compat code doesn't even work for the whole range.  We'd need to test for every version in between the range to know where it works and where it doesn't.  So even having the compat code doesn't guarantee we support versions 8.41.0+ -- we likely support 8.41.0 until some version where something changed in ESLint, and then as of this commit, we'd support 8.57.0+.

I'll happily re-add it to get this moving forward, but it won't add any practical value.

Reply:
We definitely should be testing for every minor version in between the range, regardless :-)

the practical value is that, even if untested, there are users who won't be broken by removing it.

Reply:
I suppose -- I mean, again, these were never guaranteed to be stable in the first place and have been actively breaking inside of ESLint itself in minor versions.  And the use case for moving this library without moving eslint is a bit absurd.  Again, I'm happy to add it back just to get this moving forward, but it feels very pedantic.

Reply:
Re added the compat code (although I added a slightly changed version that better matches the new 8.57.0 APIs, so that we can keep the rest of the code simpler and unaware of the backwards compat).